xdem/terrain.py

```diff
@@ -2258,3 +2283,208 @@ def fractal_roughness(
+    result = result_padded[pad_rows : pad_rows + rows, pad_cols : pad_cols + cols]
+
+    # Restore NaN values where original data was invalid
+    result[~valid_mask] = np.nan
+
+    return result
+
+
+@overload
+def texture_shading(
+    dem: NDArrayf | MArrayf,
+    alpha: float = 0.8,
+    method: str = "fft",
+    window_size: int | None = None,
+    mp_config: MultiprocConfig | None = None,
+) -> NDArrayf: ...
+
+
+@overload
+def texture_shading(
+    dem: RasterType,
+    alpha: float = 0.8,
+    method: str = "fft",
+    window_size: int | None = None,
+    mp_config: MultiprocConfig | None = None,
+) -> RasterType: ...
+
+
+def texture_shading(
+    dem: NDArrayf | MArrayf | RasterType,
+    alpha: float = 0.8,
+    method: str = "fft",
+    window_size: int | None = None,
+    mp_config: MultiprocConfig | None = None,
+) -> NDArrayf | RasterType:
+    """
+    Generate a texture shaded relief map using fractional Laplacian operator.
+
+    This technique, developed by Leland Brown, applies a fractional Laplacian operator
+    in the frequency domain to enhance terrain texture and fine-scale topographic features.
+    It's particularly effective for visualizing subtle terrain variations that may not
+    be apparent in traditional hillshading.
+
+    The fractional Laplacian operator is controlled by the alpha parameter:
+    - alpha = 0: Preserves all frequencies with mean removed (~zero-centered DEM)
+    - alpha = 1: Standard Laplacian operator (edge detection)
+    - alpha = 2: Enhanced high-frequency features
+
+    Based on: Brown, L. (2010). Texture Shading: A New Technique for Depicting Terrain Relief.
+    Workshop on Mountain Cartography, Banff, Canada.
+
+    Adapted from the Python implementation at https://github.com/fasiha/texshade-py
+
+    :param dem: Input DEM array or Raster object
+    :param alpha: Fractional exponent for Laplacian operator (0-2, default 0.8).
+        Higher values enhance fine details, lower values provide smoother results.
+    :param method: Processing method, currently only "fft" (frequency domain) is supported
+    :param window_size: Window size for local processing (None for global processing)
+    :param mp_config: Multiprocessing configuration, run the function in multiprocessing if not None
+
+    :raises ValueError: If alpha is not between 0 and 2, or if method is not supported
+
+    :examples:
+        >>> import numpy as np
+        >>> # Create a simple test DEM with a ridge
+        >>> dem = np.zeros((50, 50))
+        >>> dem[20:30, :] = np.sin(np.linspace(0, np.pi, 50)) * 10
+        >>> textured = texture_shading(dem, alpha=0.8)
+        >>> textured.shape == dem.shape
+        True
+        >>> # Higher alpha enhances fine details
+        >>> textured_enhanced = texture_shading(dem, alpha=1.5)
+
+    :returns: Texture shaded array with same shape as input DEM
+    """
+    # Get array from input
+    dem_arr = gu.raster.get_array_and_mask(dem)[0]
```

Comment:
We should be able to call `get_terrain_attribute(attribute="texture_shading")` here to avoid repeating the Raster unwrapping/wrapping :)